### PR TITLE
North Carolina Military Retirement Deduction 2022

### LINKED ISSUE #5015: North Carolina Military Retirement Deduction 2022
NC militray retirement

[Reference](https://www.ncdor.gov/2022-tax-law-changespdf-0/open)

### PR DESCRIPTION
**Pull request by vrathi101**
Fixes #5015

### CODE REVIEW

File: policyengine_us/parameters/gov/states/nc/tax/income/deductions/deductions.yaml [hunk 1]
```diff
@@ -6,6 +6,12 @@ values:
     - nc_child_deduction
     - taxable_social_security
 
+  2022-01-01:
+    - us_govt_interest
+    - nc_standard_or_itemized_deductions
+    - nc_child_deduction
+    - taxable_social_security
+    -
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/income/deductions/deductions.yaml:14]:
```suggestion
    - nc_military_retirement_deduction
```

File: policyengine_us/parameters/gov/states/nc/tax/income/deductions/military_retirement/minimum_years.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+description: North Carolina requires a filer to have served for at least this many years in the military to qualify for specific military retirement deductions, given the filer is not medically retired.
+values:
+  2022-01-01: 20
+
+metadata:
+  label: North Carolina minimum years serving in the military
+  unit: /1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/income/deductions/military_retirement/minimum_years.yaml:7]:
```suggestion
  unit: years
```

File: policyengine_us/variables/gov/states/nc/tax/income/deductions/military_retirement/nc_military_retirement_deduction_eligible.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+from policyengine_us.model_api import *
+
+
+class nc_military_retirement_deduction_eligible(Variable):
+    value_type = bool
+    entity = Person
+    label = "North Carolina military retirement deduction eligible"
+    unit = USD
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nc/tax/income/deductions/military_retirement/nc_military_retirement_deduction_eligible.py:8]:
```suggestion
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/household/demographic/person/is_medically_retired.py
Comment (MaxGhenis) [inline policyengine_us/variables/household/demographic/person/is_medically_retired.py:1]: Does this seem sufficiently different from `is_permanently_and_totally_disabled` or `is_permanently_disabled_veteran` to require a new variable?

Comment (vrathi101) [reply policyengine_us/variables/household/demographic/person/is_medically_retired.py:1]: > Does this seem sufficiently different from `is_permanently_and_totally_disabled` or `is_permanently_disabled_veteran` to require a new variable?

i guess not actually, I will use the `is_permanently_disabled_veteran` variable instead

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**